homeassistant/components/zwave/__init__.py

```diff
@@ -830,6 +830,25 @@ def __init__(self, values, domain):
 
     def network_value_changed(self, value):
         """Handle a value change on the network."""
+        if (value.command_class == const.COMMAND_CLASS_CENTRAL_SCENE and
+                value.node.node_id == self.node.node_id and
+                self.hass is not None):
+
+            scene_id = int(str(value.index) + str(value.data))
```

Comment:
Why are we doing this math? Shouldn't these be individual event attributes?

Reply:
It's not really performing math, so much as concatenating the two individual integers for the scene number and scene key that OZW is passing us in the ValueChanged notification.  For example, with the HS-WD100+ dimmer and a double tap for the on button, the ```value.index``` I get from OZW is ```1```, and the ```value.data``` I get is ```3```, which I concatenate into a ```scene_id``` of ```13```.  I was looking for a way I could pass them along separately to the event I fire and utilize that in an automation, but I didn't notice any way to do so with the existing code that fires the ZWave scene events, so this seemed like a suitable compromise to create a unique scene from both pieces of data I get from OZW.